app/assets/javascripts/initializers/initScrolling.js

```diff
@@ -121,22 +121,13 @@ function buildTagsHTML(tag) {
     antifollow +
     '</h3>' +
     '<p class="grid-cell__summary truncate-at-3"></p>' +
-    '<form class="edit_follow flex items-center flex-nowrap mb-4" id="edit_follow_' +
+    '<input name="follows[][id]" id="follow_id" type="hidden" form="follows_update_form" value="' +
```

Comment:
I think `follows[id]` should work, don't need the extra empty bracket `[]`.

Reply:
Rack will parse the empty bracket to an array (by [normalize_params](https://github.com/rack/rack/blob/1d70c73043e8fd41bf204ff794730aed2620d7a4/lib/rack/query_parser.rb#L102)), and append the following value to it.
So with `follows[][id]`, the params passed to the server will be `follows: [{id: 1}, {id: ...}, {...}]`
By doing this, the form can pass an array of hashes, which is easier for rails to handle the parameter restriction.

Here is a[ blog](http://codefol.io/posts/How-Does-Rack-Parse-Query-Params-With-parse-nested-query/) that describes this feature. Although this was post several years ago, rack remains the same mechanism.

I have fought on this trick for several hours, hope it makes sense 😅